plugins/jabra-gnp/fu-jabra-gnp-device.c

```diff
@@ -0,0 +1,704 @@
+	fu_progress_add_step(progress, FWUPD_STATUS_DEVICE_WRITE, 30, "write-chunks");
+	fu_progress_add_step(progress, FWUPD_STATUS_DEVICE_WRITE, 15, "read-verify-status");
+	fu_progress_add_step(progress, FWUPD_STATUS_DEVICE_WRITE, 15, "write-version");
+
+	/* write partition */
+	blob = fu_firmware_get_bytes(img, error);
+	if (blob == NULL)
+		return FALSE;
+	if (!fu_jabra_gnp_device_write_partition(self, fu_firmware_get_idx(img), error))
+		return FALSE;
+	fu_progress_step_done(progress);
+
+	/* start erasing? */
+	if (!fu_jabra_gnp_device_start(self, error))
+		return FALSE;
+	fu_progress_step_done(progress);
+
+	/* poll for erase done? */
+	if (!fu_jabra_gnp_device_flash_erase_done(self, error))
+		return FALSE;
+	fu_progress_step_done(progress);
+
+	/* write chunks */
+	chunks = fu_chunk_array_new_from_bytes(blob, 0x00, 0x00, chunk_size);
+	if (!fu_jabra_gnp_device_write_crc(self,
+					   fu_jabra_gnp_image_get_crc32(FU_JABRA_GNP_IMAGE(img)),
+					   chunks->len,
+					   100,
+					   error))
+		return FALSE;
+	if (!fu_jabra_gnp_device_write_chunks(self, chunks, fu_progress_get_child(progress), error))
+		return FALSE;
+	fu_progress_step_done(progress);
+
+	/* verify */
+	if (!fu_jabra_gnp_device_read_verify_status(self, error))
+		return FALSE;
+	fu_progress_step_done(progress);
+
+	/* write version */
+	if (!fu_jabra_gnp_device_write_version(
+		self,
+		fu_jabra_gnp_firmware_get_version_data(FU_JABRA_GNP_FIRMWARE(firmware)),
+		error))
+		return FALSE;
+	fu_progress_step_done(progress);
+
+	/* success */
+	return TRUE;
+}
+
+static gboolean
+fu_jabra_gnp_device_write_firmware(FuDevice *device,
+				   FuFirmware *firmware,
+				   FuProgress *progress,
+				   FwupdInstallFlags flags,
+				   GError **error)
+{
+	FuJabraGnpDevice *self = FU_JABRA_GNP_DEVICE(device);
+	g_autoptr(GPtrArray) imgs = fu_firmware_get_images(firmware);
+
+	/* progress */
+	fu_progress_set_id(progress, G_STRLOC);
+	fu_progress_add_flag(progress, FU_PROGRESS_FLAG_NO_PROFILE);
+	for (guint i = 0; i < imgs->len; i++) {
+		FuFirmware *img = g_ptr_array_index(imgs, i);
+		fu_progress_add_step(progress,
+				     FWUPD_STATUS_UNKNOWN,
+				     fu_firmware_get_size(img),
+				     fu_firmware_get_id(img));
+	}
+	for (guint i = 0; i < imgs->len; i++) {
+		FuFirmware *img = g_ptr_array_index(imgs, i);
+		if (!fu_jabra_gnp_device_write_image(self,
+						     firmware,
+						     img,
+						     fu_progress_get_child(progress),
+						     error)) {
+			g_prefix_error(error, "failed to write %s: ", fu_firmware_get_id(img));
+			return FALSE;
+		}
+		fu_progress_step_done(progress);
+	}
+	/* write squif */
+	if (!fu_jabra_gnp_device_write_dfu_from_squif(self, error))
+		return FALSE;
+	return TRUE;
+}
+
+static void
+fu_colorhug_device_set_progress(FuDevice *self, FuProgress *progress)
+{
+	fu_progress_set_id(progress, G_STRLOC);
+	fu_progress_add_flag(progress, FU_PROGRESS_FLAG_GUESSED); // FIXME, remove
```

Comment:
@gdpcastro do you get warnings in your console like https://github.com/fwupd/fwupd/wiki/Daemon-Warning:-FuProgress-steps -- if so can you submit the real values and remove GUESSED please.

Reply:
I do not get those warnings (even tried running with --verbose), which seems a bit odd, as I'm quite sure writing of chunks takes most of the time for most images -- also to note that since images have different sizes the steps would be different for each image I believe. 
I do however get a "Less than one minute remaining" message on the console when there is definitely more than one minute remaining, not sure how problematic that is.

Reply:
> I do however get a "Less than one minute remaining" message on the console when there is definitely more than one minute remaining

Right, this is probably because the values are wildly inaccurate. Can you try using `meson configure -Dsupported_build=disabled`? That's normally only set by distro packages. that have had QA.

Reply:
I tried this, it didn't seem to help. I'm trying to fidget a bit with the progress, but it doesn't seem like I know how to fix it. I entered some somewhat more accurate values for the progress of each image (generalized, as images vary in size), but I still get the "less than one minute remaining" console output way too early.